pdp/static/js/pdp_raster_map.js

```diff
@@ -126,18 +126,49 @@ function intersection(b1, b2) {
 
 function getRasterNativeProj(capabilities, layer_name) {
     // Return the dataset native projection as an OL.Projection object
-    var bbox = capabilities.find('Layer > Name:contains("' + layer_name + '")').parent().find('BoundingBox')[0],
-        srs = bbox.attributes.getNamedItem('SRS');
-    return new OpenLayers.Projection(srs.value);
+    const bboxes = capabilities.find(
+      'Capability > Layer > Layer > Layer > BoundingBox'
+    );
+    // Since the dataset <Name> is now the filepath, which does not necessarily
+    // match `layer_name`, we can't find it that way. Instead we make the
+    // somewhat fragile but currently valid assumption that the first (and
+    // only) such layer is the one we want.
+    const bbox = bboxes[0];
```

Comment:
Just to make sure I understand what's going on here, the thing we're assuming is that all layers / variables in a file have the same spatial extent, right?

Reply:
Yes. That is one way of putting it, and possibly a better one.